src/Jwt/LcobucciFactory.php

```diff
@@ -53,10 +53,14 @@ public function __construct(string $secret, string $algorithm = 'hmac.sha256', ?
         }
 
         $signerClass = self::SIGN_ALGORITHMS[$algorithm];
+        /** @phpstan-ignore-next-line */
+        $signer = is_a($signerClass, Signer\Ecdsa::class, true) ? $signerClass::create() : new $signerClass();
```

Comment:
This was a bug, ecdsa signer never worked before as the constructor is expecting an argument ( convertor ).

Reply:
phpstan still complains about it as it thinks `$signerClass` on the right side of the operation can be an Ecdsa signer ( requires 1 parameter in the constructor ), but that is false.